File: Assets/Bundles/Shared/Resources/TS/VoxelWorld/Block.lua
-- Compiled with unity-ts v2.1.0-75
local _ItemDefinitions = require("Shared/TS/Item/ItemDefinitions")
local GetItemMeta = _ItemDefinitions.GetItemMeta
local GetItemTypeFromBlockId = _ItemDefinitions.GetItemTypeFromBlockId
local Block
do
	Block = setmetatable({}, {
		__tostring = function()
			return "Block"
		end,
	})
	Block.__index = Block
	function Block.new(...)
		local self = setmetatable({}, Block)
		return self:constructor(...) or self
	end
	function Block:constructor(voxel, world)
		self.voxel = voxel
		self.world = world
		self.blockId = VoxelWorld:VoxelDataToBlockId(voxel)
		self.itemType = GetItemTypeFromBlockId(self.blockId)
		if self.itemType then
			self.itemMeta = GetItemMeta(self.itemType)
		end
	end
	function Block:IsAir()
		return self.blockId == 0
	end
	function Block:GetBlockDefinition()
		return self.world:GetBlockDefinition(self.blockId)
	end
	function Block:GetAverageColor()
		return self:GetBlockDefinition().averageColor:GetValue(0)
	end
end
return {
	Block = Block,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Shared/VoxelWorld/Block.ts
import { ItemMeta } from "Shared/Item/ItemMeta";
import { ItemType } from "Shared/Item/ItemType";
import { World } from "./World";
import { ItemUtil } from "../Item/ItemUtil";

export class Block {
	public readonly blockId: number;
	public readonly itemType: ItemType | undefined;
	public readonly itemMeta: ItemMeta | undefined;

	constructor(public readonly voxel: number, public readonly world: World) {
		this.blockId = VoxelWorld.VoxelDataToBlockId(voxel);
		this.itemType = ItemUtil.GetItemTypeFromBlockId(this.blockId);
		if (this.itemType) {
			this.itemMeta = ItemUtil.GetItemMeta(this.itemType);
		}
	}

	public IsAir(): boolean {
		return this.blockId === 0;
	}

	public GetBlockDefinition(): BlockDefinition {
		return this.world.GetBlockDefinition(this.blockId)!;
	}

	public GetAverageColor(): Color {
		return this.GetBlockDefinition().averageColor.GetValue(0);
	}
}
